File: WPF/MVVM/AddChildList.vb
Imports System.Windows.Markup

Public Class AddChildList(Of T)
    Inherits NotifyingList(Of T)
    Implements IAddChild

    Public Sub AddChild(value As Object) Implements IAddChild.AddChild
        Me.Add(DirectCast(value, T))
    End Sub

    Public Sub AddText(text As String) Implements IAddChild.AddText
        Throw New NotSupportedException()
    End Sub

End Class

File: WPF/MVVM/AddChildList.cs
using System;
using System.Windows.Markup;

namespace Ks
{
    namespace Common
    {
        public class AddChildList<T> : NotifyingList<T>, IAddChild
        {
            public void AddChild(object value)
            {
                this.Add((T)value);
            }

            public void AddText(string text)
            {
                throw new NotSupportedException();
            }
        }
    }
}
